pkg/controllers/terminating/v1alpha1/termination.go

```diff
@@ -0,0 +1,97 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	provisioning "github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/awslabs/karpenter/pkg/utils/pod"
+	"github.com/awslabs/karpenter/pkg/utils/ptr"
+
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Terminator struct {
+	kubeClient    client.Client
+	cloudprovider cloudprovider.Factory
+	coreV1Client  corev1.CoreV1Interface
+}
+
+// cordon takes in a list of expired nodes as input and cordons them
+func (t *Terminator) cordon(ctx context.Context, node *v1.Node) error {
+	persisted := node.DeepCopy()
+	node.Spec.Unschedulable = true
+	node.Labels[provisioning.ProvisionerPhaseLabel] = provisioning.ProvisionerDrainingPhase
+	if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
+		return fmt.Errorf("patching node %s, %w", node.Name, err)
+	}
+	zap.S().Debugf("Cordoned node %s", node.Name)
+	return nil
+}
+
+// terminateNodes takes in a list of expired non-drained nodes and calls drain on them
+func (t *Terminator) drain(ctx context.Context, node *v1.Node) (bool, error) {
+	if phase, ok := node.Labels[provisioning.ProvisionerPhaseLabel]; ok && phase == provisioning.ProvisionerDrainedPhase {
```

Comment:
You don't need both of these checks, if not ok, then phase will be "". You can just do `node.Labels[provisioning.ProvisionerPhaseLabel] == provisioning.ProvisionerDrainedPhase`

Reply:
Also -- looking at the code, how does this work? Is ProvisionerDrainedPhase ever set? IMO, we shouldn't even need `ProvisionerDrainedPhase`